src/pages/players/[id]-[username]/statistics.astro

```diff
@@ -0,0 +1,98 @@
+---
+export const prerender = false
+
+import {
+  Aggregation,
+  LeaderboardEntriesApi,
+  LeaderboardEntryHistoryRow,
+  PlayersApi,
+  Resolution,
+} from "../../../lib/api"
+import Section from "../../../components/layout/Section.astro"
+import Main from "../../../components/layout/Main.astro"
+import { MmrHistoryChart } from "../../../components/widgets/MmrHistoryChart"
+import { Widget } from "../../../components/ui/Widget"
+import { FactionDropdown } from "../../../components/widgets/FactionDropdown"
+import { getDataOrErrorResponse } from "../../../lib/utils"
+import Sidebar from "../../../components/layout/Sidebar.astro"
+import PlayerActivity from "../../../components/widgets/PlayerActivity.astro"
+import PlayerMatchupStats from "../../../components/widgets/PlayerMatchupStats.astro"
+import PlayerGameLengthStats from "../../../components/widgets/PlayerGameLengthStats.astro"
+import PlayerOpponents from "../../../components/widgets/PlayerOpponents.astro"
+import PlayerLayout from "../../../layouts/PlayerLayout.astro"
+
+const playerId = Astro.params.id!
+
+const [[player, playerActivity, playerMatchupStats, playerOpponents], error] = await getDataOrErrorResponse(
+  PlayersApi.getPlayer({ playerId }),
+  PlayersApi.getPlayerStatisticsActivity({ playerId }),
+  PlayersApi.getPlayerStatisticsMatchups({ playerId }),
+  PlayersApi.getPlayerStatisticsOpponents({ playerId, count: 10 })
+)
+if (error) return error
+
+const leaderboardEntries = player?.leaderboard_entries
+  .filter((x) => x.matches > 0)
+  .sort((a, b) => (b.points ?? 0) - (a.points ?? 0))
+
+const faction = new URLSearchParams(Astro.url.search).get("faction") || leaderboardEntries?.[0].race
+const leaderboard = leaderboardEntries.find((leaderboard) => leaderboard.race === faction) || leaderboardEntries?.[0]
+
+let leaderboardHistory: LeaderboardEntryHistoryRow[] = []
+
+if (leaderboard?.leaderboard_entry_id) {
+  const [[leaderboardEntryHistory], error2] = await getDataOrErrorResponse(
+    LeaderboardEntriesApi.getLeaderboardEntryHistory({
+      leaderboardEntryId: leaderboard?.leaderboard_entry_id,
+      resolution: Resolution.DAY,
+      aggregation: Aggregation.LAST,
+    })
+  )
+  leaderboardHistory = leaderboardEntryHistory.history
+  if (error2) return error
+}
```

Comment:
I did that because it depends on a first API call to get the leaderboard id and also to make sure I am not calling it with a null/undefined id.
Do you have a cleaner way to do that?